recipes/fft/all/conanfile.py

```diff
@@ -0,0 +1,91 @@
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "threads": [True, False],
+               "max_threads": "ANY",
+               "threads_begin_n": "ANY"}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "threads": False,
+                       "max_threads": 4,
+                       "threads_begin_n": 65536}
+    exports_sources = ["CMakeLists.txt", "fft_build.c", "alloc.h.fixed", "fft.h", "fft2.h", "fft3.h", "dct.h"]
+    generators = "cmake",
+
+    @property
+    def _build_subfolder(self):
+        return "_build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+        os.unlink("alloc.h")
+        os.rename("alloc.h.fixed", "alloc.h")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+    
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if not self.options.threads:
+            del self.options.max_threads
+            del self.options.threads_begin_n
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+    
+    def validate(self):
+        def _is_power_of_two(n):
+            return (n != 0) and (n & (n-1) == 0)
+
+        if self.options.threads:
+            if not self.options.max_threads.value.isdigit():
+                raise ConanInvalidConfiguration("max_threads must be an integer")
+            if not self.options.threads_begin_n.value.isdigit():
+                raise ConanInvalidConfiguration("threads_begin_n must be an integer")
+            if not _is_power_of_two(int(self.options.max_threads.value)):
+                raise ConanInvalidConfiguration("max_threads must be a power of 2")
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["THREADS"] = self.options.threads
+        if self.options.threads:
+            cmake.definitions["MAX_THREADS"] = self.options.max_threads
+            cmake.definitions["THREADS_BEGIN_N"] = self.options.threads_begin_n
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"),
+"""Copyright
+    Copyright(C) 1997,2001 Takuya OOURA (email: ooura@kurims.kyoto-u.ac.jp).
+    You may use, copy, modify this code for any purpose and 
+    without fee. You may distribute this ORIGINAL package.""")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["fft"]
+        if not self.options.shared:
+            self.cpp_info.defines.append("FFT_STATIC_DEFINE")
```

Comment:
Should be useless since you have added generate_export_header

Reply:
[generate_export_header](https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html) uses exactly that macro:
```
If the same sources are used to create both a shared and a static library, the uppercased symbol ${BASE_NAME}_STATIC_DEFINE should be used when building the static library:
```

Reply:
This macro is useful when you create both a static & shared lib in the same build, it's not an interface definition.

Reply:
I've removed `FFT_STATIC_DEFINE` from cpp_info, it works without it so far